Doc/library/asyncio-eventloop.rst

```diff
@@ -215,21 +217,21 @@ Scheduling callbacks
    Schedule the *callback* :term:`callback` to be called with
    *args* arguments at the next iteration of the event loop.
 
-   Callbacks are called in the order in which they are registered.
-   Each callback will be called exactly once.
+   Callbacks are called in the order in which they are registered. Each callback
+   will be called exactly once, even if it is registered multiple times.
 
    An optional keyword-only *context* argument allows specifying a
    custom :class:`contextvars.Context` for the *callback* to run in.
-   The current context is used when no *context* is provided.
+   Callbacks use the current context when no *context* is provided.
 
-   An instance of :class:`asyncio.Handle` is returned, which can be
+   Returns an instance of :class:`asyncio.Handle`, which can be
    used later to cancel the callback.
 
    This method is not thread-safe.
 
 .. method:: loop.call_soon_threadsafe(callback, *args, context=None)
 
-   A thread-safe variant of :meth:`call_soon`.  Must be used to
+   A thread-safe variant of :meth:`call_soon`. This method *must* be used to
    schedule callbacks *from another thread*.
```

Comment:
It was not totally clear to me whether this means to say that:
* This method must be used If _the callback_ will be run in another thread than _this method is called in_,, which is what I _think_ it is trying to say, or
* This method _must be used from another thread_ to schedule callbacks, as I first read it, or
* Something else?

Here's a suggestion trying to clarify the first option:

```suggestion
   A thread-safe variant of :meth:`call_soon`.
   This method **must** be used instead of :meth:`!call_soon` to
   schedule callbacks if they are to be called from another thread.
```

Also, used bold instead of italics to avoid confusion with the standard formatting of parameter names.

Reply:
@gvanrossum Can you provide clarity on the technical details here?

Reply:
I'd simply reverse it: "When scheduling callbacks from another thread, this function must be called, since `call_soon()` is not thread-safe."

Reply:
Taking the above and clarifying further the finer points a bit, specifically what "from another thread" refers to (another thread than the caller? The callback is executed in? Or something else?) and being more explicit that it must be used _instead of_ `call_soon`, both as above, we have


```suggestion
   A thread-safe variant of :meth:`call_soon`.
   When scheduling callbacks to be called from another thread,
   this method **must** be used instead of the non-thread-safe :meth:`!call_soon`.
```

Reply:
Yeah, though I'm less excited than CAM about the **bold**. (I feel that **bold** for emphasis is pretty extreme, and I don't think there's confusion possible with other uses of italics here.

Reply:
Another option might be [RFC2199 "MUST"](https://www.rfc-editor.org/rfc/rfc2119)?

A quick search of the codebase shows what appear to be numerous uses of RFC2119, so this seems stylistically consistent to me.

Reply:
To me that feels even stronger than the bold. If bold is too strong, then we can just use italics anyway; certainly not worth holding up this PR over:

```suggestion
   A thread-safe variant of :meth:`call_soon`.
   When scheduling callbacks to be called from another thread,
   this method *must* be used instead of the non-thread-safe :meth:`!call_soon`.
```

Reply:
Not to strain overly hard at a gnat, but --- to me, the emphasis of **bold** has a different meaning from the emphasis of MUST: Bold is a semantic emphasis, as part of an attempt to persuade the reader by reasoning; whereas MUST is a technical emphasis, indicating how one is to use the method correctly.

Given that it is critical in a technical sense for a caller to use this method and not `call_soon()` from another thread, I think the emphasis of MUST is specifically warranted here.

Reply:
That's true, but I at least strongly (heh) associate it with mandatory clauses in normative specifications, as opposed to user-facing instructions. But...meh, it's not _that_ important to me what color we paint the bikeshed... :)

Reply:
I appreciate your perspective, it's what I was going for more than anything. (Apologies to anyone watching the PR for the noise as we discussed.)

On looking back at Guido's original recommendation, I think his wording is semantically different from your recommendation, so I'm actually going to edit to match his, since he is the subject matter expert here. Commit coming shortly.